### DISCUSSION TITLE
Using `clone-in-kitty` with `open-actions.conf`

### DISCUSSION THREAD

**Discussion by ad-chaos** [kovidgoyal/kitty#5931] [Q&A]
Here's what I have in my `open-actions.conf` (copied straight from the docs)
```py
protocol file
fragment_matches [0-9]+
action launch --cwd=current --type=tab nvim +${FRAGMENT} ${FILE_PATH}
```

I would like to be able to propagate the `PATH` variable into the `nvim` process, so that it recognises the `brew` installed programs. I tried `clone-in-kitty` as so:

`action launch --cwd=current --type=tab clone-in-kitty nvim +${FRAGMENT} ${FILE_PATH}`
but it complains that:
`Failed to launch child: clone-in-kitty With error: No such file or directory`

Which is rather confusing because its supposed to launch a zsh process and zsh does exist?

I have also tried
- `action launch clone-in-kitty --type=tab --cwd=current nvim +${FRAGMENT} ${FILE_PATH}` which opens a new `split` rather than a new `tab` and fails with the same error.
- `action launch --type=tab clone-in-kitty --cwd=current nvim +${FRAGMENT} ${FILE_PATH}`
- `action launch --cwd=current clone-in-kitty --type=tab nvim +${FRAGMENT} ${FILE_PATH}` this is similar to the first variant

I may have mistaken `clone-in-kitty`'s usage and perhaps this is all just gibberish.

I also did try  `action launch --cwd=current --type=tab zsh -c "nvim +${FRAGMENT} ${FILE_PATH}"`
but that does nothing at all (probably something crashes), I tried looking at the `stderr` by launching kitty from zsh but that makes it not crash and work normally :/

I do see the `--env` option but that requires me to specify the entire variable and `--copy-env` will not reflect changes that were made by the zsh process itself. The way brew handles updating `PATH` is that it asks the user to put 
`eval "$(/opt/homebrew/bin/brew shellenv)"` into `.zprofile` and it does the rest of the magic (atleast that's what I presume).

What I mean to ask is, is there a nice way to propagate `PATH` into the child process using `open-actions`?
Thank you!

**Comment by kovidgoyal** [comment]
clone-in-kitty is a shell alias, you cannot use it anywhere other than
in your shell. It is meant to clone your shell environment into a new
window. open actions have no access to the environment of the shell,
they only get a URL. You cant pass anything other than
what's in the URL itself.

**Reply by ad-chaos** [reply]
_sigh_ That's fair.

Any idea why `action launch --cwd=current --type=tab zsh -c "nvim +${FRAGMENT} ${FILE_PATH}"` doesn't work?
I suppose the only option then is to specify the `PATH` variable manually?

**Reply by kovidgoyal** [reply]
Because zsh -c wont read all the zsh rc files. What you should really do is specify path in kitty.conf (via the env directive) not in zshrc. Then zsh will inherit it automatically inside kitty. Of course then it wont work inside other terminals/when sshing *into* this machine. The perfect solution is to convince apple to allow you to specify PATH system wide.

**Reply by ad-chaos** [reply]
> What you should really do is specify path in kitty.conf (via the env directive) not in zshrc

Oh! That is interesting. 

> The perfect solution is to convince apple to allow you to specify PATH system wide.

Would it be sane to edit `/etc/paths`?
